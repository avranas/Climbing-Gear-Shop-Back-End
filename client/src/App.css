
/*Header*/
header {
  align-items: center;
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin: 1rem;
}

.logo {
  height: 5rem;
  width: 9.5rem;
  background-image: url("./images/logo.jpg");
  background-size: cover;
}

#cart-icon {
  height: 3rem;
  width: 3rem;
  background-image: url("./images/cart.png");
  background-size: cover;
}

.header-button-spacer {
  margin: 0.25rem;
}

#simple-header {
  justify-content: center;
}

/*Splash*/

#splash {
  display: flex;
  width: 100%;
  height: 50rem;
  background-image: url("./images/splash.jpg");
  background-size: cover;
  align-items:center;
} 

#splash h1 {
  width: 100%;
  padding: 2rem;
  color: #0e0e0f;
  justify-content: center;
  text-align: center;
  font-family:'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  font-weight: 600;
  opacity: 0.9;
  background-image: linear-gradient(to bottom right, rgb(255, 255, 255), rgb(255, 252, 238));
}

h2 {
  padding: 2rem;
}

#category-list {
  display: flex;
}

.category-box p {
  margin-top: 1rem;
  text-align: center;
  font-size: 1.5rem;
  font-weight: 600;
}

.category-box img {
  margin: 1rem;
  width: 14rem;
  height: 14rem;
  padding: 2rem;
  background-image: linear-gradient(to bottom right, rgb(250, 252, 239), rgb(247, 241, 206));
  box-shadow: 0.1rem 0.1rem 0.1rem #b4b090;
  border-radius: 0.5rem;
}

footer {
  width: 100%;
  text-align: right;
}

footer p {
  justify-content: center;
  padding-right: 1rem;
}