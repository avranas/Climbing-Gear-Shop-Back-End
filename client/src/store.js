import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});