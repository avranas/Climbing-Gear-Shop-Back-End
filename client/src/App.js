import "./App.css";
import "bootstrap/dist/css/bootstrap.min.css";
import DetailedHeader from "./components/Headers/DetailedHeader";
import SimpleHeader from "./components/Headers/SimpleHeader";
import HomePage from "./routes/HomePage/HomePage";
import RegisterPage from "./routes/FormPage/RegisterPage";
import LoginPage from "./routes/FormPage/LoginPage";
import ProductListPage from "./routes/ProductListPage/ProductListPage";
import Notification from "./components/Notification/Notification";
import { useSelector } from "react-redux";
import { selectNotifications } from "./slices/notificationSlice";
import { Routes, Route, useSearchParams } from "react-router-dom";
import Logout from "./routes/Logout/Logout";
import ProductPage from "./routes/ProductPage/ProductPage";
import Cart from "./routes/CartPage/CartPage";
import Checkout from "./routes/FormPage/Checkout/Checkout";
import OrderPlaced from "./routes/OrderPlacedPage/OrderPlacedPage";
import OrdersList from "./routes/OrderListPage/OrderListPage";
import OrderDetailsPage from "./routes/OrderDetailsPage/OrderDetailsPage";
import SuccessfulLogin from "./routes/SuccessfulLogin/SuccessfulLogin";
import ProtectedRoutes from "./components/ProtectedRoute/ProtectedRoute";

function App() {
  const [searchParams] = useSearchParams();
  const notifications = useSelector(selectNotifications);

  return (
    <div className="App">
      <Routes>
        {[
          "/",
          "/product",
          "/cart",
          "/orders/:page",
          "/products/:page",
          "/product/:id",
          "/added-to-cart",
          "/cart",
          "/order-placed",
          "/order/:id",
          "successful-login",
        ].map((path) => (
          <Route key={path} exact path={path} element={<DetailedHeader />} />
        ))}
        {["/checkout", "/register", "/login", "/logout"].map((path) => (
          <Route key={path} exact path={path} element={<SimpleHeader />} />
        ))}
      </Routes>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/register" element={<RegisterPage />} />
        <Route
          path="/login"
          element={
            <LoginPage
              next={searchParams.get("next")}
              error={searchParams.get("error")}
            />
          }
        />
        <Route
          path="/products/:page"
          element={
            <ProductListPage
              search={searchParams.get("search")}
              category={searchParams.get("category")}
            />
          }
        />
        <Route path="/product/:id" element={<ProductPage />} />
        <Route path="/cart" element={<Cart />} />
        <Route element={<ProtectedRoutes />}>
          <Route path="/orders/:page" element={<OrdersList />} />
          <Route path="/logout" element={<Logout />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/order-placed" element={<OrderPlaced />} />
          <Route path="/order/:id" element={<OrderDetailsPage />} />
          <Route path="/successful-login" element={<SuccessfulLogin />} />
        </Route>
      </Routes>
      {notifications.map((i, key) => (
        <Notification message={i.message} id={i.id} key={key} />
      ))}
      <footer>
        <div id="footer-spacer"></div>
        <p>
          Made with ❤️ by <a href="https://github.com/avranas">Alex Vranas</a>
        </p>
      </footer>
    </div>
  );
}

export default App;
