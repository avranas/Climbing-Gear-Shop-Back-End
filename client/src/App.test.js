import "@testing-library/jest-dom";
import App from "./App";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "./store";
import renderer from "react-test-renderer";

const app = (
  <Provider store={store}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>
);

test("renders learn react link", () => {
  const tree = renderer.create(app).toJSON();
  expect(tree).toMatchSnapshot();
});
