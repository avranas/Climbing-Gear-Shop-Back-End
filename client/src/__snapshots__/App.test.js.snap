// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <div
      id="detailed-header-content"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="logo header-item"
          id="detailed-header-logo"
        />
      </a>
      <div
        className="header-item"
      >
        <div
          data-testid="searchbar"
          id="search-bar"
        >
          <input
            id="search-text"
            name="search"
            onChange={[Function]}
            onKeyUp={[Function]}
            type="text"
            value=""
          />
          <button
            data-testid="searchbar-button"
            id="search-button"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="header-item"
        id="welcome-message"
      >
         
      </div>
      <div
        className="header-item"
        id="header-user-buttons-container"
      >
        <div
          id="header-user-buttons"
        >
          <div />
        </div>
      </div>
      <div
        className="header-item"
      >
        <a
          href="/cart"
          onClick={[Function]}
        >
          <div
            id="cart-icon"
          />
        </a>
      </div>
    </div>
  </header>
  <main
    id="home-page"
  >
    <section
      id="splash"
    >
      <h1>
        Welcome to The Fake Climbing Gear Shop
      </h1>
    </section>
    <section
      id="shop-all-products-button-holder"
    >
      <a
        href="/products/0"
        onClick={[Function]}
      >
        <button
          className="important-button"
        >
          Shop all products
        </button>
      </a>
    </section>
    <h2
      className="container"
    >
      Shop By Category
    </h2>
    <section
      className="container d-flex justify-content-around flex-wrap"
      id="category-list"
    >
      <a
        href="/products/0?category=shoes"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(48, 53, 73)",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="category-box styled-box styled-link-box"
        >
          <img
            alt="shoes"
            src="shoes.png"
          />
          <p>
            Shoes
          </p>
        </div>
      </a>
      <a
        href="/products/0?category=harnesses"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(48, 53, 73)",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="category-box styled-box styled-link-box"
        >
          <img
            alt="harnesses"
            src="harness.png"
          />
          <p>
            Harnesses
          </p>
        </div>
      </a>
      <a
        href="/products/0?category=ropes"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(48, 53, 73)",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="category-box styled-box styled-link-box"
        >
          <img
            alt="ropes"
            src="rope.png"
          />
          <p>
            Ropes
          </p>
        </div>
      </a>
      <a
        href="/products/0?category=hardware"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(48, 53, 73)",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="category-box styled-box styled-link-box"
        >
          <img
            alt="hardware"
            src="carabiner.png"
          />
          <p>
            Hardware
          </p>
        </div>
      </a>
      <a
        href="/products/0?category=chalk"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(48, 53, 73)",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="category-box styled-box styled-link-box"
        >
          <img
            alt="chalk"
            src="chalk.png"
          />
          <p>
            Chalk
          </p>
        </div>
      </a>
    </section>
    <h2
      className="container"
    >
      About Us
    </h2>
    <section
      className="container"
      id="about-us"
    >
      <p>
        The Fake Climbing Gear Shop is an e-commerce app made by
         
        <a
          href="https://github.com/avranas"
        >
           Alex Vranas
        </a>
        . It was made by the developer to practice building full stack web apps. This site does not actually sell any climbing gear, but it can do just about everything that an e-commerce app should be capable of.
      </p>
      <p>
        For payments, this site uses Stripe, which is currently set to test mode. If you would like to see how payments are processed, enter the credit card number: "4242 4242 4242 4242" with fake data for everything else.
      </p>
      <p>
        <a
          href="https://github.com/avranas/The-Fake-Climbing-Gear-Shop"
        >
          Link to GitHub repository
        </a>
      </p>
    </section>
  </main>
  <footer>
    <div
      id="footer-spacer"
    />
    <p>
      Made with ❤️ by 
      <a
        href="https://github.com/avranas"
      >
        Alex Vranas
      </a>
    </p>
  </footer>
</div>
`;
